integration/helpers.go

```diff
@@ -0,0 +1,112 @@
+// Copyright 2022 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/google/go-cmp/cmp"
+
+	eventtypes "github.com/kinvolk/inspektor-gadget/pkg/types"
+)
+
+func iterOutput[T any](output string, normalize func(*T), compare func(entry *T) error) error {
+	if len(output) == 0 {
+		return fmt.Errorf("output is empty")
+	}
+
+	scanner := bufio.NewScanner(strings.NewReader(output))
+	for scanner.Scan() {
+		var entry T
+
+		line := scanner.Text()
+		if err := json.Unmarshal([]byte(line), &entry); err != nil {
+			return fmt.Errorf("unmarshaling line %q: %w", line, err)
+		}
+
+		// To be able to use reflect.DeepEqual and cmp.Diff, we need to
+		// "normalize" the output so that it only includes non-default values
+		// for the fields we are able to verify.
+		if normalize != nil {
+			normalize(&entry)
+		}
+
+		// It returns error when we must stop iterating.
+		if err := compare(&entry); err != nil {
+			return err
+		}
+	}
+	if err := scanner.Err(); err != nil {
+		return fmt.Errorf("parsing output: %w", err)
+	}
+
+	return nil
+}
+
+func expectOnly[T any](output string, expected *T, normalize func(*T)) error {
+	return iterOutput(output, normalize, func(entry *T) error {
+		if !reflect.DeepEqual(expected, entry) {
+			return fmt.Errorf("unexpected output entry:\n%s\n", cmp.Diff(expected, entry))
+		}
+		return nil
+	})
+}
+
+func expectAllFrom[T any](output string, expectedEntries []T, normalize func(*T)) error {
```

Comment:
This name is not clear to me. Does it mean all expectedEntries should be present on the output?

Reply:
Yes, I changed it to:

```golang
// expectListIsMatched verifies that all the entries in expectedEntries is
// matched by at least one entry in the output.
func expectListIsMatched[T any](output string, expectedEntries []*T, normalize func(*T)) error
```

WDYT?

Reply:
What about `expectListMatch`?

Reply:
Forget that, I made another suggestion in my last review.